src/models/volume.js

```diff
@@ -212,6 +231,34 @@ export default {
         yield put({ type: 'changeTagsLoading', payload: { tagsLoading: false } })
       }
     },
+    *showCreateVolumeModalBefore({
+      payload,
+    }, { call, put, all }) {
+      yield put({ type: 'showCreateVolumeModal' })
+      // TODO: longhorn manager should have an API to get all volume's snapshots or add snapshotList array in GET /v1/volumes
```

Comment:
Hi @PhanLe1010 ,
In order to show all volumes snapshot in create volume modal. Current I sends mutliple POST requests to get each volume's snapshot list.

<img width="1495" alt="Screenshot 2024-06-24 at 6 22 22 PM" src="https://github.com/longhorn/longhorn-ui/assets/5744158/6f875864-180d-4bf6-9109-e37de7f2a225">

Is it possible to have new API `GET /v1/snapshots` to get all volume's snapshots like response
```
{
  vol_name: [ {...}, {...},...],
  vol1_name : [{..},{...}]
}
```

or put snapshot list data in `GET /v1/volumes` API response. 
That can save extra API call and save line 238 ~ 253
cc @derekbit

Reply:
Thanks for advice. That make sense. Modified in 2fed9e8, the behavior change like below screen video

https://github.com/longhorn/longhorn-ui/assets/5744158/bfe78da3-ac14-4ca1-9453-7755388b06ab